src/Stratis.Features.FederatedPeg/Distribution/RewardClaimer.cs

```diff
@@ -63,11 +63,11 @@ private void OnBlockConnected(BlockConnected blockConnected)
 
             // As this runs on the mainchain we presume there will be a coinstake transaction in the block (but during the PoW era there obviously may not be).
             // If not, just do nothing with this block.
-            if (maturedBlock.Transactions.Count < 2 || !blockConnected.ConnectedBlock.Block.Transactions[1].IsCoinStake)
+            if (maturedBlock.Transactions.Count < 2 || !maturedBlock.Transactions[1].IsCoinStake)
```

Comment:
Is it possible that `maturedBlock` can be `null` ?

Reply:
@quantumagi it looks like we do a check above to see if the tip is less than the `minStakeConfirmations`. This will prevent a null ChainedHeader.